src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
+
+  for (const a of rows) {
+    e.push([...a.cells]);
+  }
+
+  for (let i = 0; i < gameSize; i++) {
+    for (let x = 0; x < gameSize; x++) {
+      const gameValue = e[i][x];
+      const arrayValue = gameSquare[i][x];
+
+      gameValue.innerText = arrayValue === 0 ? '' : arrayValue;
+      gameValue.classList.value = '';
+
+      gameValue.classList.add('field-cell',
+        `field-cell--${gameValue.innerText}`);
+    }
+  }
+}
+
+function newGame() {
+  gameSquare = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  score = 0;
+  scoreField.innerText = score;
+  generateNewNumber();
+  fillGameField();
+}
+
+document.addEventListener('keydown', (e) => {
+  const pressedKey = e.code;
+
+  switch (pressedKey) {
+    case 'ArrowRight':
+      moveRowsRight();
+      generateNewNumber();
+      break;
+    case 'ArrowLeft':
+      moveRowsLeft();
+      generateNewNumber();
+      break;
+    case 'ArrowUp':
+      moveRowsUp();
+      generateNewNumber();
+      break;
+    case 'ArrowDown':
+      moveRowsDown();
+      generateNewNumber();
+      break;
+  }
+
+  fillGameField();
+
+  if (isGameOver()) {
+    loseMessage.classList.remove('hidden');
+  };
+
+  isGameWon();
+});
+
+function moveCellsLeft(array) {
+  function removeZeros(collection) {
+    return collection.filter((item) => item !== 0);
+  }
+
+  let copy = [...array];
+
+  copy = removeZeros(copy);
+
+  for (let i = 0; i < copy.length - 1; i++) {
+    if (copy[i] === copy[i + 1]) {
+      copy[i] *= 2;
+      copy[i + 1] = 0;
+      score += copy[i];
+      scoreField.innerText = score;
+    }
+  }
+
+  copy = removeZeros(copy);
+
+  while (copy.length < gameSize) {
+    copy.push(0);
+  }
+
+  return copy;
+}
+
+function moveRowsLeft() {
+  for (let i = 0; i < gameSize; i++) {
+    gameSquare[i] = moveCellsLeft(gameSquare[i], gameSize);
+  }
+
+  return gameSquare;
+}
+
+function moveRowsRight() {
+  for (let i = 0; i < gameSize; i++) {
+    gameSquare[i].reverse();
+    gameSquare[i] = moveCellsLeft(gameSquare[i], gameSize);
+    gameSquare[i].reverse();
+  }
+
+  return gameSquare;
+}
+
+function moveRowsUp() {
+  for (let i = 0; i < gameSize; i++) {
+    let newRow = [gameSquare[0][i],
+      gameSquare[1][i],
+      gameSquare[2][i],
+      gameSquare[3][i]];
+
+    newRow = moveCellsLeft(newRow, gameSize);
+
+    gameSquare[0][i] = newRow[0];
+    gameSquare[1][i] = newRow[1];
+    gameSquare[2][i] = newRow[2];
+    gameSquare[3][i] = newRow[3];
+  };
+
+  return gameSquare;
+}
+
+function moveRowsDown() {
+  for (let i = 0; i < gameSize; i++) {
+    let newRow = [gameSquare[0][i],
+      gameSquare[1][i],
+      gameSquare[2][i],
+      gameSquare[3][i]];
+
+    newRow.reverse();
+    newRow = moveCellsLeft(newRow, gameSize);
+    newRow.reverse();
+
+    gameSquare[0][i] = newRow[0];
+    gameSquare[1][i] = newRow[1];
+    gameSquare[2][i] = newRow[2];
+    gameSquare[3][i] = newRow[3];
+  };
+
+  return gameSquare;
+}
```

Comment:
These two functions look very similar, merge the repeating code into a helper function